docs/concepts/endpoint.md

```diff
@@ -0,0 +1,117 @@
+---
+id: endpoint
+title: Endpoint
+---
+
+Endpoints in ZIO HTTP are defined using the `Endpoint` object's combinators, which provide a type-safe way to specify various aspects of the endpoint. For instance, consider defining endpoints for retrieving user information and user posts:
+
+```scala mdoc:silent
+import zio._
+import zio.http._
+import zio.http.endpoint.{Endpoint, EndpointExecutor, EndpointLocator, EndpointMiddleware}
+import zio.http.codec.{HttpCodec, PathCodec}
+import HttpCodec.query
+
+val auth = EndpointMiddleware.auth
+
+val getUser = Endpoint(Method.GET / "users" / int("userId")).out[Int]
+
+val getUserPosts = Endpoint(Method.GET / "users" / int("userId") / "posts" / int("postId"))
+  .query(query("name"))
+  .out[List[String]] @@ auth
+```
+
+In these examples, we use combinators like `Method.GET`, `int`, and `query` to define the HTTP method, URL path, path parameters, query parameters, and response types of the endpoints.
+
+### Middleware Application
+
+Middleware can be applied to endpoints using the `@@` operator to add additional behavior or processing. For example, we can apply authentication middleware to restrict access to certain endpoints:
+
+```scala mdoc:silent
+
+val getUserRoute =
+  getUser.implement {
+    Handler.fromFunction[Int] { id =>
+      id
+    }
+  }
+```
+
+Here, the `auth` middleware ensu authenticated users can access the `getUser` endpoint.
+
+### Endpoint Implementation
+
+Endpoints are implemented using the `implement` method, which takes a function specifying the logic to handle the request and generate the response. Inside the implementation function, ZIO effects can be used to perform computations and interact with dependencies:
+
+```scala
```

Comment:
Please make sure to include the `mdoc` modifier in all code snippets whenever possible.

Reply:
sure